tests/benchmarks/cli/commands/test_plots.py

```diff
@@ -0,0 +1,68 @@
+from textwrap import dedent
+
+import pytest
+
+
+@pytest.fixture
+def repo_with_plots(tmp_dir, scm, dvc):
+    num_files = 20
+    CODE = dedent(
+        """
+    from random import random
+    import sys
+
+    num_files = {num_files}
+    linspace_size = 1000
+
+    def load_params():
+        import yaml
+        with open('params.yaml') as fd:
+                return yaml.safe_load(fd)
+
+    params = load_params()
+    p = params['p']
+
+    def calc_metric(index):
+        metric = [(i/linspace_size)**p + (random()-0.5)/index
+                  for i in range(1,linspace_size)]
+
+        result=[]
+        for elem in list(metric):
+            result.append({{f'val_{{index}}': elem}})
+        return result
+
+    import json
+    if __name__ == '__main__':
+        for i in range(1, num_files+1):
+            metric = calc_metric(i)
+            with open(f'metric_{{i}}.json', 'w') as fd:
+                json.dump(metric, fd)
+    """
+    ).format(num_files=num_files)
+
+    tmp_dir.gen("train.py", CODE)
+    tmp_dir.gen("params.yaml", "p: 1")
+    scm.add(["train.py", "params.yaml"])
+    scm.commit("initial")
+
+    dvc.run(
+        name="generate_plots",
+        deps=["train.py"],
+        plots=[f"metric_{i}.json" for i in range(1, num_files + 1)],
+        params=["p"],
+        cmd="python train.py",
+    )
+
+    scm.add(["dvc.lock", "dvc.yaml"])
+    scm.commit("initial run, p=1")
+
+    tmp_dir.gen("params.yaml", "p: 3")
+    dvc.reproduce()
+
+    yield
+
+
+def test_plots_diff(tmp_dir, bench_dvc, repo_with_plots):
+    bench_dvc("plots", "diff")
```

Comment:
Btw, the issue can be narrowed down to `plots show`, right? Or is there something `diff` specific that we are interested as well in?

I'm just looking at the test file generation and it is pretty slow and being run at every test run (not even cached across session or something), so wondering if we could maybe generate it once, put into our data (maybe even two versions for `diff`) and just use it here?

Reply:
> Btw, the issue can be narrowed down to plots show, right? Or is there something diff specific that we are interested as well in?

Well, when using diff we also check branched to some extent, but I wouldn't say its necessary.

> Also, do we really need that many metrics files? 

Well this depends - the problem with dpath was about retrieved data dict (the result of `Repo.plots.show`) - the bigger the dict the more visible the problem is. We could, for example create single file with few tens of thousands of points (which would show the problem with `dpath`) but for future testing I would presume that more realistic scenario would be having multiple plots files.

> Also, any other potential tests (other commands?) where these files would be useful?

Probably not.


I will change the data generation and change `diff` to `show`

Reply:
@pared Sounds good. Make sure you are using `s3://dvc-public/remote/dvc-bench` remote to upload the files (this is where `bench_s3_public` leads to). Probably need to adjust `bench_s3` to point to it, because currently it still points to the legacy place where we used to upload results with asv. I adjusted locally, but didn't push yet 🤷